### PR TITLE
fix: ensure onConnect is always called

### PR DESCRIPTION
**Pull request by ronag**
## Status

- [ ] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [ ] Tested
- [ ] Benchmarked (**optional**)
- [ ] Documented
- [ ] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by metcoder95 — COMMENTED** [review]
LGTM, but it seems it will require adjust some testing

**Comment by ronag** [comment]
@metcoder95 ptal

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/core/request.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ const { headerNameLowerCasedRecord } = require('./constants')
 const invalidPathRegex = /[^\u0021-\u00ff]/
 
 const kHandler = Symbol('handler')
+const noop = () => {}
```
Comment (ronag) [inline lib/core/request.js:29]:
```suggestion
```

File: lib/core/request.js [hunk 2]
```diff
@@ -296,6 +297,7 @@ class Request {
     }
     this.aborted = true
 
+
```
Comment (ronag) [inline lib/core/request.js:300]:
```suggestion
```

**Review by metcoder95 — APPROVED** [review]

**Pull request merged**